redfish-core/lib/cable.hpp

```diff
@@ -73,6 +126,175 @@ inline void
+                                                    updatedAssemblyList) {
+        std::string downstreamResource =
+            cableObjectPath + "/downstream_resource";
+        linkAssociatedCable(
+            asyncResp, downstreamResource,
+            [asyncResp,
+             updatedAssemblyList](const std::vector<std::string>& value) {
+                nlohmann::json& linkArray =
+                    asyncResp->res.jsonValue["Links"]["DownstreamResources"];
+                linkArray = nlohmann::json::array();
+
+                for (const auto& fullPath : value)
+                {
+                    auto it = find(updatedAssemblyList.begin(),
+                                   updatedAssemblyList.end(), fullPath);
+
+                    // If element was found
+                    if (it != updatedAssemblyList.end())
+                    {
+
+                        uint index =
+                            static_cast<uint>(it - updatedAssemblyList.begin());
+                        linkArray.push_back(
+                            {{"@odata.id", "/redfish/v1/Chassis/chassis/"
+                                           "Assembly#/Assemblies/" +
+                                               std::to_string(index)}});
+                    }
+                    else
+                    {
+                        BMCWEB_LOG_ERROR
+                            << "in Downstream Resources " << fullPath
+                            << " isn't found in chassis assembly list";
+                    }
+                }
+            });
+    };
+
+    std::string upstreamConnector = cableObjectPath + "/upstream_connector";
+    linkAssociatedCable(
+        asyncResp, upstreamConnector,
+        [asyncResp](const std::vector<std::string>& value) {
+            nlohmann::json& linkArray =
+                asyncResp->res.jsonValue["Links"]["UpstreamPorts"];
+            linkArray = nlohmann::json::array();
+
+            for (const auto& fullPath : value)
+            {
+                sdbusplus::message::object_path path(fullPath);
+                std::string leaf = path.filename();
+                if (leaf.empty())
+                {
+                    continue;
+                }
+                std::string endpointLeaf = path.parent_path().filename();
+                if (endpointLeaf.empty())
+                {
+                    continue;
+                }
+
+                //  insert/create link using endpointLeaf
+                endpointLeaf += "/Ports/";
+                endpointLeaf += leaf;
+                linkArray.push_back(
+                    {{"@odata.id",
+                      "/redfish/v1/Systems/system/FabricAdapters/" +
+                          endpointLeaf}});
+            }
+        });
+
+    redfish::chassis_utils::getChassisAssembly(asyncResp, "chassis",
+                                               std::move(chassisAssociation));
+
```

Comment:
Maybe better if moved to line number 189 OR has to be called doe after population upstream_connector?